airflow/settings.py

```diff
@@ -183,6 +183,19 @@ def configure_orm(disable_connection_pool=False):
     # For Python2 we get back a newstr and need a str
     engine_args['encoding'] = engine_args['encoding'].__str__()
 
+    # Sane timeouts for connections and statements
+    try:
+        connect_timeout_seconds = conf.getint('core', 'SQL_ALCHEMY_CONNECT_TIMEOUT_SECONDS')
+    except conf.AirflowConfigException:
+        connect_timeout_seconds = 10
+    engine_args['connect_timeout'] = connect_timeout_seconds
+
+    try:
+        statement_timeout_seconds = conf.getint('core', 'SQL_ALCHEMY_STATEMENT_TIMEOUT_SECONDS')
+    except conf.AirflowConfigException:
+        statement_timeout_seconds = 10
```

Comment:
maybe statement timeout should be a bit bigger. sometimes we're reading a lot of state at once from postgres?

Reply:
this is a default; we can bump the default to something like 180?

Reply:
The thinking is that this should always be fast; if it's not super fast then something might be super wrong. We should never be pulling out more than a few KB of data.